src/main/java/uk/gov/hmcts/reform/juddata/camel/util/JrdUserProfileUtil.java

```diff
@@ -0,0 +1,271 @@
+        var userProfilesWithInvalidPersonalCodes = filterByPersonalCode(userProfiles);
+
+        var invalidRecordsList = new ArrayList<JudicialUserProfile>();
+        Stream.of(userProfilesWithInvalidObjectIds, userProfilesWithInvalidPersonalCodes)
+                .forEach(invalidRecordsList::addAll);
+
+        return List.copyOf(invalidRecordsList);
+    }
+
+    public void remove(List<JudicialUserProfile> userProfilesToBeDeleted, List<JudicialUserProfile> userProfiles) {
+        var erroneousUserProfileSet = new HashSet<>(userProfilesToBeDeleted);
+        userProfiles.removeIf(erroneousUserProfileSet::contains);
+    }
+
+    public void audit(List<JudicialUserProfile> invalidUserProfiles, Exchange exchange) {
+        var routeProperties = (RouteProperties) exchange.getIn().getHeader(ROUTE_DETAILS);
+        FileStatus fileStatus = getFileDetails(exchange.getContext(), routeProperties.getFileName());
+
+        if (nonNull(invalidUserProfiles) && !CollectionUtils.isEmpty(invalidUserProfiles)) {
+            auditInvalidUserProfiles(invalidUserProfiles, routeProperties.getTableName());
+
+            fileStatus.setAuditStatus(PARTIAL_SUCCESS);
+            registerFileStatusBean(applicationContext, routeProperties.getFileName(), fileStatus,
+                    exchange.getContext());
+        }
+    }
+
+    private void auditInvalidUserProfiles(List<JudicialUserProfile> invalidUserProfiles, String tableName) {
+        var def = new DefaultTransactionDefinition();
+        def.setName("Jsr exception logs");
+        def.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRED);
+        Map<String, String> globalOptions = camelContext.getGlobalOptions();
+        String schedulerTime = globalOptions.get(SCHEDULER_START_TIME);
+
+        jdbcTemplate.batchUpdate(
+                invalidJsrSql,
+                invalidUserProfiles,
+                10,
+                new ParameterizedPreparedStatementSetter<>() {
+                    public void setValues(PreparedStatement ps, JudicialUserProfile argument) throws SQLException {
+                        ps.setString(1, tableName);
+                        ps.setTimestamp(2, new Timestamp(Long.parseLong(schedulerTime)));
+                        ps.setString(3, globalOptions.get(SCHEDULER_NAME));
+                        ps.setString(4, argument.getPerId());
+                        ps.setString(5, PER_CODE_OBJECT_ID_FIELD);
+                        ps.setString(6, PER_CODE_OBJECT_ID_ERROR_MESSAGE);
+                        ps.setTimestamp(7, new Timestamp(new Date().getTime()));
+                        ps.setLong(8, argument.getRowId());
+                    }
+                });
+
+        TransactionStatus status = platformTransactionManager.getTransaction(def);
+        platformTransactionManager.commit(status);
+    }
+
+    public int sendEmail(List<JudicialUserProfile> userProfilesDeleted, List<JudicialUserProfile> userProfiles) {
```

Comment:
I think this is being called only in this class - can it be private?

Reply:
why you want to test the return value as it does not do anything in here - https://github.com/hmcts/rd-judicial-data-load/blob/db6e3b649c750918506e56d66c3678a35f80882d/src/main/java/uk/gov/hmcts/reform/juddata/camel/util/JrdUserProfileUtil.java#L99

Reply:
thanks @souravroy2021 @ishratt . I have approved the changes